pkg/client/event_deduplicator.go

```diff
@@ -0,0 +1,188 @@
+package client
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	chain "github.com/keep-network/keep-ecdsa/pkg/chain"
+)
+
+// eventDeduplicator decides whether the given event should be handled by the
+// client or not.
+//
+// Event subscription may emit the same event two or more times. The same event
+// can be emitted right after it's been emitted for the first time. The same
+// event can also be emitted a long time after it's been emitted for the first
+// time. It is eventDeduplicator's responsibility to decide whether the given
+// event is a duplicate and should be ignored or if it is not a duplicate and
+// should be handled.
+//
+// Four events are supported:
+// - key generation request for a new keep,
+// - signature request for a keep,
+// - keep close request,
+// - keep terminate request.
+type eventDeduplicator struct {
```

Comment:
Thoughts on extracting `event_deduplicator.go` and `event_tracking.go` to a separate package `event`? This way we could remove the `event` prefix from file and struct names and achieve better encapsulation.

Reply:
That's not a bad idea but I'd do it as `pkg/client/event` and not as `pkg/event`. This code is about how the client reacts to those events and not about a generic concept of an event.

Reply:
Please see 7f6dd64fe65abe7f2f343dce1d873ba585cf22c3.